### PR TITLE
Replace macros by include files (#236)

### PR DESCRIPTION
**Pull request by ioioioio**
These changes replace the huge macros by include files.

The new amalgamation is a bit weird. It supposes that the special include files are all in the include folder, plus it requires that every files end by an empty line, otherwise the last line will be skipped.

### CODE REVIEW

**Comment by lemire** [comment]
@ioioioio looks like hard work!

@jkeiser : any thoughts?

File: include/simdjson/stringparsing_common.h [hunk 1]
```diff
@@ -0,0 +1,91 @@
+// This file contains the common code every implementation uses
+// It is intended to be included multiple times and compiled multiple times
+// We assume the file in which it is include already includes
+// "simdjson/stringparsing.h" (this simplifies amalgation)
+
+#ifdef TARGETED_ARCHITECTURE
```
Comment (jkeiser) [inline include/simdjson/stringparsing_common.h]: Given that it's an error to include these files without setting TARGETED_ARCHITECTURE, I'd treat this more like argument validation and switch it to #ifndef / #error. This is in the nitpick arena (isn't a bug, and is just a defensive measure), so feel free to ignore :)

```suggestion
#ifndef TARGETED_ARCHITECTURE
#error TARGETED_ARCHITECTURE must be set before including this file.
#endif
```
Then the #endif at the end of the file wouldn't be needed. Not sure how to add that part as a suggestion here.

**Comment by jkeiser** [comment]
I tend to use the Review button on things when I look through the whole thing, don't mind me :)

**Comment by jkeiser** [comment]
@ioioio a note: everything compiles fine on VS2019, but in in the IDE if you actually open the .h file, VS gets super confused and spits out a bunch of errors as if the file is being included all by itself and isn't nested inside the namespace.

I wouldn't block; if I can come up with a good solution and it gets in the way a lot, I'll send a PR down the road. The obvious solution is to put the namespace {} back into the .h file, but that means we'd have to pass 2 vars instead of one, and I rather liked that change you made.

**Comment by lemire** [comment]
@ioioioio Looks like good work. Merge at will!

**Comment by jkeiser** [comment]
I made [this PR](https://github.com/lemire/simdjson/pull/253) against this branch, which fixes the Intellisense errors by moving the namespace. We can discuss, and we don't have to block this PR on it, either :)

**Pull request merged**